timemachine/cpp/src/context.hpp

```diff
@@ -28,6 +28,15 @@ class Context {
 
     std::array<std::vector<double>, 3> multiple_steps_U(const int n_steps, int store_u_interval, int store_x_interval);
 
+    std::array<std::vector<double>, 2> multiple_steps_local(
+        const int n_steps,
+        const std::vector<int> &local_idxs,
+        const int store_x_interval,
+        const double radius,
+        const double k,
+        const double temperature,
```

Comment:
Curious if it would make sense to parameterize this instead by the ratio `k / temperature`? (assuming that e.g. doubling both `k` and `temperature` here has no effect on the selection probabilities). Would this sidestep the need to check that the temperature here is consistent with the integrator's temperature?

Reply:
Sorry, disregard the suggestion of parameterizing with k / temperature instead of (k, temperature) - I'd misunderstood that temperature wasn't required except for this ratio.

Related question though: would it make sense to remove temperature as a parameter and just use the integrator temperature?

Reply:
>Related question though: would it make sense to remove temperature as a parameter and just use the integrator temperature?

(👍 , this seems cleaner/safer, if possible)

Reply:
>If we remove temperature as a parameter, when you use an integrator (not a thermostat, IE Verlet) would it be alright to use a default temperature that is not tunable?

Ahh, I think that shows a possible inconsistency in allowing to pair "Local MD" with a "raw VelocityVerlet" integrator.

In this context I assumed we were treating "VelocityVerlet" as an approximately correct MCMC move for a specific NVT ensemble (i.e. it _is_ implicitly associated with a temperature in this context).

One correct way to use VelocityVerlet as an MCMC move would be to use [an HMC move](https://github.com/proteneer/timemachine/pull/787). ("Randomly sample velocities (depending on temperature/masses), integrate with velocity verlet (depending on masses), accept or reject (depending on temperature/masses).")

In practice this correct implementation could be close to a less explicit approximate implementation. (E.g. if the temperature is always the default temperature, initial velocities are distributed as expected given temperature, and stepsize etc. are small enough that nearly 100% of moves would be accepted, then "raw `VelocityVerlet(dt, masses)`" could have pretty similar behavior to "`HMC(dt, masses, temperature)`".)

Agnostic whether "it's the user's responsibility" to understand this and make sure masses/temperatures/initial velocities etc. are self-consistent + validate that the stepsize is small enough that approximation errors are negligible + ..., or if "it's the implementer's responsibility" to put guardrails here.

Reply:
Talking to @maxentile was decided to require a thermostat, and we will wrap the Verlet integrator into a thermostat at a later date.

https://github.com/proteneer/timemachine/pull/915/commits/0165635690f13fbc12093e1bfa0fbb3dc406afbf

Reply:
(for posterity: a consistent alternative to giving VelocityVerlet a temperature is to expose a slightly different local MD interface, as in https://github.com/proteneer/timemachine/pull/1005 )